test/router_ics_skf_worker_SUITE.erl

```diff
@@ -90,287 +75,159 @@ end_per_testcase(TestCase, Config) ->
+    {ok, JoinedDevice1} = router_device_cache:get(?CONSOLE_DEVICE_ID),
+    JoinedSKF1 = device_to_skf("route_id", JoinedDevice1),
 
-    [{Type4, Req4}] = rcv_loop([]),
-    ?assertEqual(update, Type4),
-    ?assertEqual(add, Req4#iot_config_session_key_filter_update_req_v1_pb.action),
+    %% List to make sure we have the new device
+    {ok, Remote1} = router_ics_skf_worker:remote_skf(),
+    ?assertEqual(lists:sort([JoinedSKF1 | SKFs]), lists:sort(Remote1)),
 
-    {ok, JoinedDevice1} = router_device_cache:get(?CONSOLE_DEVICE_ID),
-    <<JoinedDevAddr1:32/integer-unsigned-big>> = lorawan_utils:reverse(
-        router_device:devaddr(JoinedDevice1)
-    ),
+    %% Send first so we can rejoin and get a new devaddr
+    ct:print("sending packet for devaddr lock in"),
+    send_unconfirmed_uplink(Stream1, PubKeyBin1, JoinedDevice1),
 
-    ?assertEqual(
-        #iot_config_session_key_filter_v1_pb{
-            oui = router_utils:get_oui(),
-            devaddr = JoinedDevAddr1,
-            session_key = erlang:binary_to_list(
-                binary:encode_hex(router_device:nwk_s_key(JoinedDevice1))
-            )
-        },
-        Req4#iot_config_session_key_filter_update_req_v1_pb.filter
-    ),
+    %% Join device again
+    ct:print("joining 2"),
+    #{stream := Stream2, pubkey_bin := PubKeyBin2} = test_utils:join_device(Config),
+    {ok, JoinedDevice2} = router_device_cache:get(?CONSOLE_DEVICE_ID),
+    JoinedSKF2 = device_to_skf("route_id", JoinedDevice2),
 
-    %% Send first packet nothing should happen
-    Stream1 !
-        {send,
-            test_utils:frame_packet(
-                ?UNCONFIRMED_UP,
-                PubKeyBin1,
-                router_device:nwk_s_key(JoinedDevice1),
-                router_device:app_s_key(JoinedDevice1),
-                0
-            )},
+    ct:print("sending packet for devaddr lock in"),
+    send_unconfirmed_uplink(Stream2, PubKeyBin2, JoinedDevice2),
 
-    [] = rcv_loop([]),
+    %% List to make sure old devaddr was removed, and new one was added
+    {ok, Remote2} = router_ics_skf_worker:remote_skf(),
+    ?assertEqual(lists:sort([JoinedSKF2 | SKFs]), lists:sort(Remote2)),
 
-    %% Join device again
-    #{
-        stream := Stream2,
-        pubkey_bin := PubKeyBin2
-    } = test_utils:join_device(Config),
+    ok.
 
-    [{Type6, Req6}] = rcv_loop([]),
-    ?assertEqual(update, Type6),
-    ?assertEqual(add, Req6#iot_config_session_key_filter_update_req_v1_pb.action),
+reconcile_multiple_updates_test(_Config) ->
+    %% When more than 1 request needs to be made to the config service, we don't
+    %% know want to know until all have requests have succeeded or failed.
 
-    {ok, JoinedDevice2} = router_device_cache:get(?CONSOLE_DEVICE_ID),
-    <<JoinedDevAddr2:32/integer-unsigned-big>> = lorawan_utils:reverse(
-        router_device:devaddr(JoinedDevice2)
-    ),
+    ok = application:set_env(router, test_udpate_skf_delay_ms, 500),
```

Comment:
```suggestion
    ok = application:set_env(router, test_update_skf_delay_ms, 500),
```

Reply:
I have switched the `p` and `d` keys on my keyboard so I will never make this typo again.